src/Files.App/Helpers/DynamicDialogFactory.cs

```diff
@@ -57,50 +57,27 @@ public static DynamicDialog GetFor_ShortcutNotFound(string targetPath)
 
 		public static DynamicDialog GetFor_RenameDialog()
 		{
-			DynamicDialog dialog = null;
-			TextBox inputText = new TextBox()
+			DynamicDialog? dialog = null;
+			TextBox inputText = new()
 			{
 				Height = 35d,
 				PlaceholderText = "RenameDialogInputText/PlaceholderText".GetLocalizedResource()
 			};
 
-			TextBlock tipText = new TextBlock()
+			TextBlock tipText = new()
 			{
 				Text = "RenameDialogSymbolsTip/Text".GetLocalizedResource(),
 				Margin = new Microsoft.UI.Xaml.Thickness(0, 0, 4, 0),
 				TextWrapping = Microsoft.UI.Xaml.TextWrapping.Wrap,
 				Opacity = 0.0d
 			};
 
-			inputText.BeforeTextChanging += async (textBox, args) =>
+			inputText.TextChanged += (textBox, args) =>
 			{
-				if (FilesystemHelpers.ContainsRestrictedCharacters(args.NewText))
-				{
-					args.Cancel = true;
-					await inputText.DispatcherQueue.EnqueueAsync(() =>
-					{
-						var oldSelection = textBox.SelectionStart + textBox.SelectionLength;
-						var oldText = textBox.Text;
-						textBox.Text = FilesystemHelpers.FilterRestrictedCharacters(args.NewText);
-						textBox.SelectionStart = oldSelection + textBox.Text.Length - oldText.Length;
-						tipText.Opacity = 1.0d;
-					});
-				}
-				else
-				{
-					dialog.ViewModel.AdditionalData = args.NewText;
-
-					if (!string.IsNullOrWhiteSpace(args.NewText))
-					{
-						dialog.ViewModel.DynamicButtonsEnabled = DynamicDialogButtons.Primary | DynamicDialogButtons.Cancel;
-					}
-					else
-					{
-						dialog.ViewModel.DynamicButtonsEnabled = DynamicDialogButtons.Cancel;
-					}
-
-					tipText.Opacity = 0.0d;
-				}
+				tipText.Opacity = FilesystemHelpers.ContainsRestrictedCharacters(inputText.Text) ? 1.0d : 0.0d;
```

Comment:
x:Load instead?

Reply:
It was used with opacity previously, that's why I decided to leave it that way.

Reply:
How would you use x:load here (sorry I am quite new to xaml)

Reply:
We can create a Boolean and bind the x:Load property of the text to the property.

Reply:
But there is no actual xaml for the text, is there?

Reply:
I think we can do this with Visibility if you prefer but as there is no xaml I don't think we can use x:Load

Reply:
Also, if we make the textblock unload or make it disappear, the dialog box size changes abruptly, and I don't think that looks good

Reply:
Forgot about the size changing, let's leave as is then.